### PR TITLE
Add pre-deployed HC contracts to genesis file

### PR DESCRIPTION
**Pull request by boyuan-chen**
This PR integrates the HC contracts to the pre-deployed addresses.

- The `BobeTuringCredit` is pre-deployed at `0x42000000000000000000000000000000000003e8`.
- The `BobaHCHelper` is pre-deployed at `0x42000000000000000000000000000000000003E9`

Other changes:

- Our BOBA contract is deployed at L1 and the L2 BOBA token is pre-deployed at `0x4200000000000000000000000000000000000023` for `make devnet-up`. This makes us two methods of starting the devnet (`make devnet-up` and `make devnet-hardhat-up`) generates the same L2 network with our BOBA token.
- A new test is added for `make devnet-up` which bridges our BOBA token from L1 to L2 
- Add new contracts called `EAS` and `SchemaRegistry` to `boba-chain-ops`
- `BobaGasPriceOracle` is removed from the system
- Make some simple changes to the smart contracts.
These two variables are changed to `public`.
https://github.com/bobanetwork/v3-anchorage/blob/4d07b16b78c1f78c9f385de68a5a52e33ea12fbb/packages/contracts-bedrock/src/boba/BobaHCHelper.sol#L24-L25
The pre-deployed addresses are fixed.
https://github.com/bobanetwork/v3-anchorage/blob/4d07b16b78c1f78c9f385de68a5a52e33ea12fbb/packages/contracts-bedrock/src/boba/BobaHCHelper.sol#L13
https://github.com/bobanetwork/v3-anchorage/blob/4d07b16b78c1f78c9f385de68a5a52e33ea12fbb/packages/contracts-bedrock/src/boba/BobaHCHelper.sol#L16
https://github.com/bobanetwork/v3-anchorage/blob/4d07b16b78c1f78c9f385de68a5a52e33ea12fbb/packages/contracts-bedrock/src/boba/BobaTuringCredit.sol#L157
```
bedrock-devnet/devnet/__init__.py
bedrock-devnet/hardhat/__init__.py
boba-bindings/artifacts.json
boba-bindings/bindings/boba.go
boba-bindings/bindings/boba_gaspriceoracle.go
boba-bindings/bindings/boba_gaspriceoracle_more.go
boba-bindings/bindings/boba_more.go
boba-bindings/bindings/bobahchelper.go
boba-bindings/bindings/bobahchelper_more.go
boba-bindings/bindings/bobaturingcredit.go
boba-bindings/bindings/bobaturingcredit_more.go
boba-bindings/bindings/governancetoken.go
boba-bindings/bindings/l2governanceerc20.go
boba-bindings/bindings/mips_more.go
boba-bindings/bindings/preimageoracle_more.go
boba-bindings/gen/main.go
boba-bindings/predeploys/addresses.go
boba-chain-ops/README.md
boba-chain-ops/cmd/check-l2/main.go
boba-chain-ops/genesis/check.go
boba-chain-ops/genesis/config.go
boba-chain-ops/genesis/layer_two_test.go
boba-chain-ops/genesis/setters.go
boba-chain-ops/immutables/immutables.go
boba-chain-ops/immutables/immutables_test.go
op-bindings/artifacts.json
op-bindings/bindings/boba.go
op-bindings/bindings/boba_more.go
op-bindings/bindings/bobahchelper.go
op-bindings/bindings/bobahchelper_more.go
op-bindings/bindings/bobaturingcredit.go
op-bindings/bindings/bobaturingcredit_more.go
op-bindings/bindings/l2governanceerc20.go
op-bindings/bindings/l2governanceerc20_more.go
op-bindings/bindings/mips_more.go
op-bindings/bindings/preimageoracle_more.go
op-bindings/bindings/registry.go
op-bindings/gen/main.go
op-bindings/predeploys/addresses.go
op-chain-ops/cmd/check-l2/main.go
op-chain-ops/genesis/config.go
op-chain-ops/genesis/layer_two_test.go
op-chain-ops/genesis/setters.go
op-chain-ops/immutables/immutables.go
op-chain-ops/immutables/immutables_test.go
op-chain-ops/state/state.go
packages/contracts-bedrock/scripts/Deploy.s.sol
packages/contracts-bedrock/scripts/Deployer.sol
packages/contracts-bedrock/src/boba/BOBA.sol
packages/contracts-bedrock/src/boba/BobaHCHelper.sol
packages/contracts-bedrock/src/boba/BobaTuringCredit.sol
packages/contracts-bedrock/src/boba/Boba_GasPriceOracle.sol
packages/sdk/tasks/deposit-boba.ts
```

TODO:
1. Add a script in boba-chain-ops to migrate storage from the old BobeTuringCredit to the new one.
2. Add a script to adapt boba-chain-ops for the Ethereum mainnet L2.

These two changes will be created in separate PRs, as this PR is already too large.

### CODE REVIEW

File: boba-chain-ops/README.md [hunk 1]
```diff
@@ -452,3 +452,10 @@ We can restore the legacy system by resetting the addresses in `Lib_AddressManag
   	"Proxy__L1StandardBridge"
   }
   ```
+
+# Hybrid Compute
+
+We will move the `BobeTuringCredit` contract from address `0x4200000000000000000000000000000000000020` to `0x42000000000000000000000000000000000003e8`. The new `BobaHCHelper` contract will be pre-deployed at `0x42000000000000000000000000000000000003E9`. The legacy storage of `BobaTuringHelper` at `0x4200000000000000000000000000000000000021` will be destroyed and replaced by the standard proxy contract.
```
Comment (jyellick) [inline boba-chain-ops/README.md]: Nit: spelling `Bobe` vs `Boba`

File: packages/contracts-bedrock/src/boba/BobaTuringCredit.sol [hunk 2]
```diff
@@ -149,4 +149,28 @@ contract BobaTuringCredit {
 
         IERC20(turingToken).safeTransfer(owner, _withdrawAmount);
     }
+
+    // ----------------------------------------------------
+    // New for Anchorage
+
+    // Could make this configurable, but for now it's hard-coded to a predeploy address
```
Comment (jyellick) [inline packages/contracts-bedrock/src/boba/BobaTuringCredit.sol:156]: If we made this configurable, would it give us more flexibility for what degree of EVM vs. Application space the implementation lands -- or are we generally protected by the ability to upgrade the contract itself?

**Review by jyellick — APPROVED** [review]
Overall LGTM

File: packages/contracts-bedrock/src/boba/BobaTuringCredit.sol (hunk 2 above)
Comment (boyuan-chen) [reply packages/contracts-bedrock/src/boba/BobaTuringCredit.sol:156]: @mmontour1306 Any thoughts about this?

Comment (mmontour1306) [reply packages/contracts-bedrock/src/boba/BobaTuringCredit.sol:156]: The BobaTuringCredit contract is only needed for legacy support, and at present I can't think of any cases where we would need to make the HCHelper address configurable. Even if we do move some of the implementation logic out of the Sequencer we would still want to have the Helper contract deployed to a known address.

**Review by mmontour1306 — APPROVED** [review]

File: boba-chain-ops/README.md (hunk 1 above)
Comment (boyuan-chen) [reply boba-chain-ops/README.md]: Fixed

**Pull request merged**